client/allocrunner/taskrunner/connect_native_hook.go

```diff
@@ -190,6 +206,25 @@ func (h *connectNativeHook) tlsEnv(env map[string]string) map[string]string {
 	return m
 }
 
+// bridgeEnv creates a set of additional environment variables to be used when launching
+// the connect native task. This will enable the task to communicate with Consul
+// if the task is running inside an alloc's network namespace (i.e. bridge mode).
+//
+// Sets CONSUL_HTTP_ADDR if not already set.
```

Comment:
Why would it already be set? By users who want to override it?

Reply:
Yeah, if someone manually sets (via env/template) any of the standard consul environment variables I believe those values should always take priority. Maybe they have Consul bound to an addressable IP?

Reply:
@shoenig I hope it is okay to ask here, before opening a ticket. The PR nicely adds this for bridge mode, but in `host` mode it still requires special configuration if eg the consul port of the local agent is changed (not sure if that makes sense to do, but…). Is there any reason to not also map this socket in in `host` network mode -- or at least provide `CONSUL_HTTP_ADDR` with the addr that nomad knows for connecting to consul? Or is that done somewhere else and I missed it?

Reply:
Hi @apollo13 ! The socket dance exists in the bridge hook because that's how we get "around" the network namespace and access Consul - which is typically listening on a loopback interface not privy to the isolated task. A Connect native task using host networking OTOH shouldn't have a problem connecting to Consul directly, and so no sockets are made for them. Your suggestion of automatically setting `CONSUL_HTTP_ADDR` is probably reasonable - feel free to open an issue/PR to discuss more.